### DISCUSSION TITLE
how to get a QueryRef without actually issuing the query, relying on future refetch calls?

### DISCUSSION THREAD

**Discussion by andreimatei** [the-guild-org/apollo-angular#2057] [Q&A]
I'm struggling to figure out how to create an Observable that tracks the results to a query without actually issuing the query (because I don't know the arguments I want to use yet).

Here's what I'm trying to do. I'm generally quite new to Angular, so if I'm holding everything wrong, I'd love to hear.
Say I have a parent component and a child component. The child component wants to take in an `Observable` as an `@Input({required: true})`. The parent component would like to bind the child input to one of the parent's fields - in particular an Observable that tracks the results of a query.
Generally, I believe one does this by creating the respective observable in the parent's `ngOnInit()`. With `apollo-angular`, one gets an observable by creating a `queryRef` with `queryRef = queryB.watch(args)`. The `queryRef` can be turned into an observable through its `valueChanges` property. And one can then also call `queryRef.refetch(newArgs)` to re-issue the query and get new results.

Now, my problem is that, at `ngOnInit` time, I don't yet have the arguments for my `queryB`. I first need another `queryA` to finish. So, the question is - is there any way I can get a `queryRef/Observable` that will be hooked to future `refetch()` calls, but without actually issuing an initial query?

**Comment by PowerKiKi** [comment]
You are most likely looking for https://rxjs.dev/api/operators/switchMap, so you can do something that look like this pseudo-code:

```ts
queryRefA = queryA.watch(args);
resultB = queryRefA.valueChanges.pipe(switchMap(resultA => {
    return queryB.watch(resultA).valueChanges;
}))

resultB.subscribe(); // start by doing XHR for A, when A returns, starts XHR for B, and return result for B
```
